uportal-war/src/main/java/org/apereo/portal/security/provider/cas/CasAssertionSecurityContext.java

```diff
@@ -167,6 +200,23 @@ protected void copyAssertionAttributesToUserAttributes(Assertion assertion) {
             log.debug("Adding attribute '{}' from Assertion with value '{}'; runtime type of value is {}",
                     attrEntry.getKey(), attrEntry.getValue(), attrEntry.getValue().getClass().getName());
 
+            // Check for credential
+            if (decryptCredentialToPassword && cipher != null && attrEntry.getKey().equals(CREDENTIAL_KEY)) {
+                try {
+                    final String encPwd = (String) (attrEntry.getValue() instanceof List
+                            ? ((List) attrEntry.getValue()).get(0) : attrEntry.getValue());
+                    byte[] cred64 = Base64.getDecoder().decode(encPwd);
```

Comment:
For uPortal 5, it could be nice to keep the more modern Java 8 version.
For uPortal 4 `javax.xml.bind.DatatypeConverter` has similar functionality.

ref: http://stackoverflow.com/a/14413290